src/main/java/me/ryanhamshire/GPFlags/flags/FlagDef_ViewContainers.java

```diff
@@ -0,0 +1,108 @@
+package me.ryanhamshire.GPFlags.flags;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.WeakHashMap;
+
+import org.bukkit.Bukkit;
+import org.bukkit.block.Block;
+import org.bukkit.block.BlockState;
+import org.bukkit.block.Container;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.block.Action;
+import org.bukkit.event.inventory.InventoryClickEvent;
+import org.bukkit.event.inventory.InventoryCloseEvent;
+import org.bukkit.event.player.PlayerInteractEvent;
+import org.bukkit.inventory.Inventory;
+
+import me.ryanhamshire.GPFlags.Flag;
+import me.ryanhamshire.GPFlags.FlagManager;
+import me.ryanhamshire.GPFlags.GPFlags;
+import me.ryanhamshire.GPFlags.MessageSpecifier;
+import me.ryanhamshire.GPFlags.Messages;
+import me.ryanhamshire.GriefPrevention.Claim;
+import me.ryanhamshire.GriefPrevention.ClaimPermission;
+import me.ryanhamshire.GriefPrevention.GriefPrevention;
+import me.ryanhamshire.GriefPrevention.PlayerData;
+
+/**
+ * Allow players to view (but not manipulate) containers (chests, dispensers, hoppers etc) within a claim.
+ * Example use case: Player casino's to verify odds.
+ * 
+ * @author Lewys Davies (Lew_)
+ */
+public class FlagDef_ViewContainers extends FlagDefinition {
+    
+    // WeakHashSet
+    private final static Set<Inventory> viewing = Collections.newSetFromMap(new WeakHashMap<Inventory, Boolean>()); 
+    
+    public FlagDef_ViewContainers(FlagManager manager, GPFlags plugin) {
+        super(manager, plugin);
+    }
+    
+    @EventHandler
+    public void onInvOpen(PlayerInteractEvent event) {
+        if(event.getAction() != Action.RIGHT_CLICK_BLOCK) return;
+        if(event.getClickedBlock() == null) return; // Just in case
+        
+        Block block = event.getClickedBlock();

```

Comment:
this could go above the null check
Suggested change:
```java
Block block = event.getClickedBlock();
if (block == null) return;
```

Reply:
I have updated the code to fit the project's coding style. 

I can confirm the flag has been thoroughly tested:
- All clicks in both inventories (The copy & players)
- Viewing an inventory whilst an authorised player is editing the original. As expected the copy doesnt adjust realtime and is completely seperate.
- Works with all container types.

This update is live on minesuperior.com and as such has been rigorously tested by our team.

As a bonus the "you don't have permission to use that" message also displays. I intentionally kept this :)

Let me know if you require anything else,
Lewys